src/middleware/identity/userIdentity.js

```diff
@@ -3,12 +3,21 @@
 const errors = require('../../errors');
 
 module.exports = function (req, res, next) {
-    if (req.identity.type !== 'User' && req.identity.type !== 'ServiceAccount') {
-        return next(new errors.Forbidden('Supplied identity not valid for requested operation'));
-    }
+    switch(req.identity.type) {
+        case 'User':
+            if (!req.identity?.user?.username || req.identity?.user?.is_internal === undefined) {
+                return next(new errors.Forbidden('Supplied identity invalid'));
+            }
+            break;
+
+        case 'ServiceAccount':
+            if (!req.identity?.service_account?.username) {
```

Comment:
Is there a place elsewhere where the username is retrieved from the header? This function only checks the header.